### PR TITLE
Add hud-set-xstr sexp

### PR DESCRIPTION
**Pull request by MjnMixael**
There is precedent in the sexp system for xstr specific sexps with modify-variable-xstr, tech-add-intel-xstr, tech-remove-intel-xstr, etc. This adds hud-set-xstr, a version of hud-set-text, allowing to assign a string and id to a hud gauge while also performing key token and variable translation.

I also took the liberty of adding key token translation to hud-set-message because it really probably should be, but I'm not hard set on that... but I think anyone using $alt-j$ in their message text probably really expects that to be translated to the bound key.

### CODE REVIEW

File: code/parse/sexp.cpp [hunk 1]
```diff
@@ -13064,6 +13065,36 @@ void sexp_hud_set_text(int n)
 	}
 }
 
+void sexp_hud_set_xstr(int n)
+{
+	bool is_nan, is_nan_forever;
+
+	auto gaugename = CTEXT(n);
+	auto text = CTEXT(CDR(n));
+	auto id = eval_num(CDR(n), is_nan, is_nan_forever);
```
Comment (wookieejedi) [inline code/parse/sexp.cpp:13074]: Just to check, I reckon perhaps `id` should use `CDDR(n)` instead of `CDR(n)` (otherwise it's the same as `text`)

Comment (MjnMixael) [reply code/parse/sexp.cpp:13074]: Correct. Fixed. Good catch.

File: code/parse/sexp.cpp [hunk 2]
```diff
@@ -682,6 +682,7 @@ SCP_vector<sexp_oper> Operators = {
 	{ "hud-set-custom-gauge-active",	OP_HUD_SET_CUSTOM_GAUGE_ACTIVE,			2, 	INT_MAX, 	SEXP_ACTION_OPERATOR,	},
 	{ "hud-set-builtin-gauge-active",	OP_HUD_SET_BUILTIN_GAUGE_ACTIVE,		2, 	INT_MAX,	SEXP_ACTION_OPERATOR,	},
 	{ "hud-set-text",					OP_HUD_SET_TEXT,						2,	2,			SEXP_ACTION_OPERATOR,	},	//WMCoolmon
+	{ "hud-set-xstr",                   OP_HUD_SET_XSTR,                        3,  3,          SEXP_ACTION_OPERATOR,   },  //MjnMixael
```
Comment (wookieejedi) [inline code/parse/sexp.cpp:685]: Thanks for the quick fix! Just one last thing I was pondering, it looked like all the other `-xstr` related sexps have `-xstr` appended to the end of the original name, so I was just wondering what your thoughts on doing the same here (ie 
having the name be `hud-set-text-xstr`)?

Comment (MjnMixael) [reply code/parse/sexp.cpp:685]: I'll be honest, my reasons are entirely selfish... using hud-set-xstr will in-place replace all of BtA's Lua `hud-set-xstr` SEXPs that turned out to be buggy which is why an in-engine solution was preferred.

Comment (wookieejedi) [reply code/parse/sexp.cpp:685]: Just as a potential thought another way is to use the deprecation string replace in `get_sexp()` (like `remove-weapons` and `clear-weapons`)

Anyhow, I'm always a sucker for internal consistency such as using `hud-set-text-xstr` but certainly won't hold up the PR over it :D

Comment (MjnMixael) [reply code/parse/sexp.cpp:685]: I think there's competing consistencies here then. `sexp-xstr` is one, sure. But all the `hud-set` sexps are `hud-set-type` and Text is distinct from Xstr. For the latter, `hud-set-xstr` is consistent with the other hud sexps.

Comment (wookieejedi) [reply code/parse/sexp.cpp:685]: Oh yeah that makes sense, thanks for the extra discussion!

**Review by wookieejedi — APPROVED** [review]

**Pull request merged**